src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/BaseRepository.cs

```diff
@@ -0,0 +1,108 @@
+namespace Altinn.Platform.Storage.Repository
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Altinn.Platform.Storage.Configuration;
+    using Altinn.Platform.Storage.Extensions;
+    using Microsoft.Azure.Documents;
+    using Microsoft.Azure.Documents.Client;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>
+    /// Base repository service for initializing db, collections etc.
+    /// </summary>
+    internal abstract class BaseRepository : IHostedService, IDisposable
+    {
+        private readonly string _collectionId;
+        private readonly string _partitionKey;
+        private readonly AzureCosmosSettings _cosmosSettings;
+
+        /// <summary>
+        /// Database name
+        /// </summary>
+        protected string DatabaseId { get; private set; }
+
+        /// <summary>
+        /// Database uri.
+        /// </summary>
+        protected Uri DatabaseUri { get; private set; }
+
+        /// <summary>
+        /// Collection uri.
+        /// </summary>
+        protected Uri CollectionUri { get; private set; }
+
+        /// <summary>
+        /// The DocumentClient.
+        /// </summary>
+        protected DocumentClient Client { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BaseRepository"/> class.
+        /// </summary>
+        /// <param name="collectionId">The ID of the collection.</param>
+        /// <param name="partitionKey">The PK of the collection.</param>
+        /// <param name="cosmosSettings">The settings object.</param>
+        public BaseRepository(string collectionId, string partitionKey, IOptions<AzureCosmosSettings> cosmosSettings)
+        {
+            _collectionId = collectionId;
+            _partitionKey = partitionKey;
+            _cosmosSettings = cosmosSettings.Value;
+
+            DatabaseId = _cosmosSettings.Database;
+            DatabaseUri = UriFactory.CreateDatabaseUri(DatabaseId);
+            CollectionUri = UriFactory.CreateDocumentCollectionUri(DatabaseId, collectionId);
+        }
+
+        /// <inheritdoc/>
+        public async Task StartAsync(CancellationToken cancellationToken)
```

Comment:
I can't see that this method is ever referenced?

Reply:
The class implements `IHostedService`, which have a lifecycle that is controlled by the host. You can see the specific implementation of this here: 
https://github.com/dotnet/runtime/blob/18dc0203bd4b2ff813bcde1f808dffc9bce391a6/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L60-L71

Here are the relevant docs: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-5.0&tabs=visual-studio

so the use cases for `IHostedService`/`BackgroundService`  include anything that revolves around initialization/cleanup where you want the host lifecycle management to take care of it